python/ray/tune/syncer.py

```diff
@@ -864,6 +871,13 @@ def wait_for_all(self):
                 f"trials to sync: \n{sync_str}"
             )
 
+    def on_experiment_end(self, trials: List["Trial"], **info):
+        """Wait for background sync processes to finish on experiment end."""
+        try:
+            self.wait_for_all()
+        except TuneError as e:
+            logger.error(e)
```

Comment:
This will provide the error message for all syncs that failed (see the `wait_for_all` raise).

For regular usage of `wait_for_all`, we may want to switch to just logging an error. In that case, we can remove this try catch.